src/layers/microscopy-viewer-layer/microscopy-viewer-layer-base.js

```diff
@@ -40,51 +38,42 @@ const defaultProps = {
 
 export class MicroscopyViewerLayerBase extends BaseTileLayer {
   constructor(props) {
-    const { sliderValues, colorValues, channelsOn } = props;
-    const orderedChannelNames = Object.keys(sliderValues);
-    const { orderedSliderValues, orderedColorValues } = setOrderedValues(
-      orderedChannelNames,
-      colorValues,
-      sliderValues,
-      channelsOn
+    const { sliderValues, colorValues, channelsOn, loader } = props;
+
+    const lengths = [sliderValues.length, colorValues.length];
+    if (lengths.every(l => l !== lengths[0])) {
+      throw Error('Inconsistent number of slider values and colors provided');
+    }
+
+    const colors = colorValues.map((color, i) =>
+      channelsOn[i]
+        ? color.map(c => c / MAX_COLOR_INTENSITY)
+        : DEFAULT_COLOR_OFF
+    );
+
+    const sliders = sliderValues.map((slider, i) =>
+      channelsOn[i] ? slider : DEFAULT_SLIDER_OFF
     );
 
     // Need to pad sliders and colors with default values (required by shader)
-    const padSize = MAX_SLIDERS_AND_CHANNELS - orderedChannelNames.length;
+    const padSize = MAX_SLIDERS_AND_CHANNELS - colors.length;
     if (padSize < 0) {
       throw Error('Too many channels specified for shader.');
     }
     const paddedSliderValues = padWithDefault(
-      orderedSliderValues,
+      sliders,
       DEFAULT_SLIDER_OFF,
       padSize
     );
+
     const paddedColorValues = padWithDefault(
-      orderedColorValues,
+      colors,
       DEFAULT_COLOR_OFF,
       padSize
     );
 
-    const getZarr = ({ x, y, z }) => {
-      return loadZarr({
-        x,
-        y,
-        z: -1 * z,
-        ...props
-      });
-    };
-    const getTiff = ({ x, y, z }) => {
-      return loadTiff({
-        x,
-        y,
-        z: -1 * z,
-        ...props
-      });
-    };
-    const getTileData =
-      (props.useZarr && getZarr) ||
-      (props.useTiff && getTiff) ||
-      props.getTileData;
+    const getTileData = ({ x, y, z }) =>
+      loader.getTile({ x, y, z: -z, ...props });
```

Comment:
This is because deck.gl's convention for z is different than most of the rest of the world?

Reply:
yes